src/colvarbias_opes.cpp

```diff
@@ -0,0 +1,1898 @@
+// This code is mainly adapted from the PLUMED opes module, which uses the
+// LGPLv3 license as shown below:
+/* +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+   Copyright (c) 2020-2021 of Michele Invernizzi.
+
+   This file is part of the OPES plumed module.
+
+   The OPES plumed module is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   The OPES plumed module is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with plumed.  If not, see <http://www.gnu.org/licenses/>.
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ */
+
+#include "colvarbias_opes.h"
+#include "colvarbias.h"
+#include "colvardeps.h"
+#include "colvarproxy.h"
+#include "colvars_memstream.h"
+#include "colvargrid.h"
+
+#include <exception>
+#include <iomanip>
+#include <ios>
+#include <stdexcept>
+#include <algorithm>
+#include <numeric>
+#include <unordered_set>
+#include <limits>
+
+colvarbias_opes::colvarbias_opes(char const *key):
+  colvarbias(key), m_kbt(0), m_barrier(0), m_biasfactor(0),
+  m_bias_prefactor(0), m_temperature(0),
+  m_pace(0), m_adaptive_sigma_stride(0),
+  m_adaptive_counter(0), m_counter(1),
+  m_compression_threshold(0), m_compression_threshold2(0),
+  m_adaptive_sigma(false), m_fixed_sigma(false),
+  m_no_zed(false), m_nlist(false), m_recursive_merge(true),
+  m_nlist_param(2, 0), m_epsilon(0), m_sum_weights(0),
+  m_sum_weights2(0), m_cutoff(0), m_cutoff2(0),
+  m_zed(1), m_old_kdenorm(0), m_kdenorm(0),
+  m_val_at_cutoff(0), m_nlist_center(0), m_nlist_index(0),
+  m_nlist_steps(0), m_nlist_update(false),
+  m_nlist_pace_reset(false), m_nker(0), m_calc_work(false),
+  m_work(0), comm(single_replica), m_num_walkers(1),
+  m_num_threads(1), m_nlker(0), m_traj_output_frequency(0),
+  m_traj_line(traj_line{0}), m_is_first_step(true),
+  m_pmf_grid_on(false), m_reweight_grid(nullptr),
+  m_pmf_grid(nullptr), m_pmf_hist_freq(0), m_pmf_shared(true)
+{
+}
+
+int colvarbias_opes::init(const std::string& conf) {
+  int error_code = colvarbias::init(conf);
+  enable(f_cvb_scalar_variables);
+  get_keyval_feature(this, conf, "applyBias", f_cvb_apply_force, true);
+  m_temperature = cvm::proxy->target_temperature();
+  m_kbt = m_temperature * cvm::proxy->boltzmann();
+  get_keyval(conf, "pace", m_pace);
```

Comment:
This looks more like a PLUMED parameter name. For Colvars, I suggest something like kernelUpdateFrequency. The same applies to the other parameters.

Reply:
Well, I just assumed that users are familiar with OPES via the PLUMED tutorials. I can change the naming style later.